### ISSUE TITLE
[Bug]: Accessibility support in the PDF.js web viewer (web/viewer.html)

### ISSUE THREAD

**Issue by Karthikvenkat86** [mozilla/pdf.js#20114]
### Attach (recommended) or Link to PDF file

I’m assessing accessibility support in the PDF.js web viewer (web/viewer.html). Currently, all text—including headers—is rendered as plain elements, without semantic roles like role="heading" or aria-level. Are there UI settings I’m missing, or is this a known limitation/bug in the viewer?

As of now, its generates a Header where screen reader cannot identify this as a header.

<img width="1923" height="948" alt="Image" src="https://github.com/user-attachments/assets/8794af84-a24c-45b8-934e-d202d5777616" />

### Web browser and its version

Edge

### Operating system and its version

Windows 11

### PDF.js version

v5.3.93

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

Generate a browser version of PDF and header doesn't have proper roles.

https://matrix.to/#/!NofDvGEWAARHltVRTj:mozilla.org/$-0tsHB06haPW340J8k8XD_HcJd0879uftHG-LhtN-C0?via=mozilla.org&via=matrix.org&via=igalia.com

<img width="3642" height="1986" alt="Image" src="https://github.com/user-attachments/assets/9f09ce9f-92af-4540-a7cd-2af80d30ee58" />

### What is the expected behavior?

Need proper accessiblity label and roles for every element.

If its header then role=header, if its table then role="table"

### What went wrong?

No proper roles

### Link to a viewer

its private

**Comment by menturion** [comment]
Same here, v5.3.93.

**Comment by Karthikvenkat86** [comment]
Can we please prioritize this bug?

**Comment by calixteman** [comment]
@Karthikvenkat86, is this bug about the viewer by itself ? or about the generated html used to render the text layer ?

**Comment by Snuffleupagus** [comment]
Closing as INCOMPLETE, given that no PDF document was provided and the lack of response to the questions in https://github.com/mozilla/pdf.js/issues/20114#issuecomment-3338374136.
